.github/workflows/pytest.yml

```diff
@@ -74,42 +74,28 @@ jobs:
         run: pip install pytest-github-actions-annotate-failures
 
       - name: Run PyTest and Generate Coverage Report
+        # with the "set -o pipefail" the "pytest" exit code won't be suppressed by "tee" command
         run: |
-          pytest -n 4 -m "not cuda" --junitxml=pytest.xml \
-          --cov-report=term-missing:skip-covered --cov=mrpro | tee pytest-coverage.txt
+          (set -o pipefail && pytest -n 4 -m "not cuda" --junitxml=pytest.xml \
```

Comment:
https://github.com/actions/runner-images/issues/4459#issuecomment-1374859960

setting shell to bash also enables  pipefail

Reply:
https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#exit-codes-and-error-action-preference